src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -535,6 +576,44 @@ public AsyncConceptChangeBatch getConceptBulkChange(@PathVariable String branch,
 		return findConceptsWithECL(">" + conceptId, form == Relationship.CharacteristicType.stated, branch, acceptLanguageHeader, 0, LARGE_PAGE.getPageSize()).getItems();
 	}
 
+	@GetMapping(value = "/browser/{branch}/concepts/ancestorPaths")
+	@JsonView(value = View.Component.class)
+	public Collection<ConceptMini> findConceptAncestorPaths(@PathVariable String branch,
+			@RequestParam(required = false) List<Long> conceptIds,		
+			@RequestParam(defaultValue = "inferred") Relationship.CharacteristicType form,
+			@RequestHeader(value = "Accept-Language", defaultValue = Config.DEFAULT_ACCEPT_LANG_HEADER) String acceptLanguageHeader) {
+
+		branch = BranchPathUriUtil.decodePath(branch);	
+		
+		Map<String, ConceptMini> conceptMiniMap = conceptService.findConceptMinis(branch, conceptIds, ControllerHelper.parseAcceptLanguageHeaderWithDefaultFallback(acceptLanguageHeader)).getResultsMap();
+		
+		// For each concept, lookup a single ancestor-path from it to the top-level concept, and add the path to the result output.
+		Collection<ConceptMini> conceptsWithAncestorPaths = new ArrayList<>();
+		
+		for(final String conceptId : conceptMiniMap.keySet()) {
+			ArrayList<ConceptMini> ancestorPath = ancestorPathHelper(branch, form, conceptId, new ArrayList(), acceptLanguageHeader);
+			conceptMiniMap.get(conceptId).addExtraField("descriptions", conceptMiniMap.get(conceptId).getActiveDescriptions());
+			conceptMiniMap.get(conceptId).addExtraField("ancestorPath", ancestorPath);
+			conceptsWithAncestorPaths.add(conceptMiniMap.get(conceptId));
+		}
+		
+		return conceptsWithAncestorPaths;
+	}	
+	
+	private ArrayList<ConceptMini> ancestorPathHelper(String branch, Relationship.CharacteristicType form, String conceptId, ArrayList<ConceptMini> pathSoFar, String acceptLanguageHeader) {
+		Collection<ConceptMini> conceptParents = findConceptParents(branch, conceptId, form, false, acceptLanguageHeader);
+		
+		if(conceptParents.isEmpty()) {
+			return pathSoFar;
+		}
+		else {
+			ConceptMini conceptParent = conceptParents.iterator().next();
```

Comment:
A concept can have multiple parents. This method will randomly return "a" path, but not "the" path because in many cases "the" path can not be represented by a single array.

Reply:
If this is understood and acceptable for your application please make this limitation explicit in the API annotation notes (which become visible in the swagger page).

Reply:
This was the intention - to choose a single path for the list view, rather than trying to calculate all possible paths (they can navigate up the other paths in the taxonomy view).

Adding notes to state this.